src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final String WRONG_INDEX_MSG = "Invalid index. Check the list size";
+    public static final String NO_SUCH_ELEMENT = "Can't find such element";
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array;
+    private int currentSize;
+    private int maxSize;
+
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_SIZE];
+        maxSize = INITIAL_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        sizeCheck();
+        array[currentSize] = value;
+        currentSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > currentSize) {
+            throw new ArrayListIndexOutOfBoundsException(WRONG_INDEX_MSG);
+        }
+        sizeCheck();
+        T[] buffer = (T[]) new Object[currentSize - index];
+        System.arraycopy(array, index, buffer, 0, buffer.length);
+        array[index] = value;
+        System.arraycopy(buffer, 0, array, index + 1, buffer.length);
```

Comment:
we can swap 34 and 35 lines, so we don't need `buffer` variable and can use `array` here